src/keypair.rs

```diff
@@ -49,6 +49,24 @@ impl From<SecretKey> for Keypair {
     }
 }
 
+impl<'a> TryFrom<&'a [u8; KEYPAIR_LENGTH]> for Keypair {
+    type Error = SignatureError;
+
+    fn try_from(bytes: &'a [u8; KEYPAIR_LENGTH]) -> Result<Self, Self::Error> {
+        let (secret, public) = split_array_ref::<
+            KEYPAIR_LENGTH, SECRET_KEY_LENGTH, PUBLIC_KEY_LENGTH
+        >(bytes);
+        let secret = SecretKey::from(secret);
+        let public = PublicKey::try_from(public)?;
+
+        if public != (&secret).into() {
+            Err(InternalError::MismatchedKeypairError.into())
+        } else {
+            Ok(Keypair { secret, public })
+        }
+    }
+}
```

Comment:
This is copy/paste of `Keypair::from_bytes`:

https://github.com/dalek-cryptography/ed25519-dalek/blob/9638ab4/src/keypair.rs#L94-L110

At the very least one should be composed in terms of the other.

But also as I [discussed in #225](https://github.com/dalek-cryptography/ed25519-dalek/issues/225#issuecomment-1322358913), we should probably get rid of the 64-byte keypair APIs, or at least deprecate them, as they don't follow the RFC8032 definition of a secret key and include redundant information which provides no additional value and only permits potential misuse.

Reply:
> At the very least one should be composed in terms of the other.

They are.  This commit changes from_bytes to check length and then call try_from.

I actually don’t care about Keypair, but I figured that if I’m adding the conversion to SecretKey and PublicKey it makes sense for completeness to have it in Keypair as well.